app/src/ui/app-error.tsx

```diff
@@ -91,6 +94,51 @@ export class AppError extends React.Component<IAppErrorProps, IAppErrorState> {
     }, dialogTransitionLeaveTimeout)
   }
 
+  private renderErrorWithMetaDataFooter(error: ErrorWithMetadata) {
+    const { retryAction } = error.metadata
+    if (retryAction !== undefined) {
+      if (retryAction.type === RetryActionType.Clone) {
+        return this.renderRetryCloneFooter(retryAction)
+      }
+    }
+
+    if (isGitError(error.underlyingError)) {
+      return this.renderGitErrorFooter(error.underlyingError)
+    }
+
+    return this.renderDefaultFooter()
+  }
+
+  private renderRetryCloneFooter(retryAction: RetryAction) {
+    let retryTitle = 'Retry'
+
+    if (this.isCloneError) {
```

Comment:
Can we get rid of this check and assume that this function is only called when there is a clone error? (its name is  `renderRetryCloneFooter()` so it should be a safe assumption).